homeassistant/components/telegram_bot/__init__.py

```diff
@@ -857,131 +858,102 @@ def leave_chat(self, chat_id=None):
+        _LOGGER.debug("Handling update %s", update)
+        if not self.authorize_update(update):
+            return
+
+        # establish event type: text, command or callback_query
+        if update.callback_query:
+            # NOTE: Check for callback query first since effective message will be populated with the message
+            # in .callback_query (python-telegram-bot docs are wrong)
+            event_type, event_data = self._get_callback_query_event_data(
+                update.callback_query
+            )
+        elif update.effective_message:
+            event_type, event_data = self._get_message_event_data(
+                update.effective_message
+            )
+        else:
+            _LOGGER.warning("Unhandled update: %s", update)
             return True
-        if ATTR_CALLBACK_QUERY in data:
-            event = EVENT_TELEGRAM_CALLBACK
-            data = data.get(ATTR_CALLBACK_QUERY)
-            message_ok, event_data = self._get_message_data(data)
-            if event_data is None:
-                return message_ok
 
-            query_data = event_data[ATTR_DATA] = data[ATTR_DATA]
+        _LOGGER.debug("Firing event %s: %s", event_type, event_data)
+        self.hass.bus.async_fire(event_type, event_data)
+        return True
+
+    @staticmethod
+    def _get_command_event_data(command_text: str):
+        if not command_text.startswith("/"):
+            return {}
+        command_parts = command_text.split()
+        command = command_parts[0]
+        args = command_parts[1:]
+        return {ATTR_COMMAND: command, ATTR_ARGS: args}
+
+    def _get_message_event_data(self, message: Message):
+        event_data: dict[str, Any] = {
+            ATTR_MSGID: message.message_id,
+            ATTR_CHAT_ID: message.chat.id,
+        }
+        if Filters.command.filter(message):
+            # This is a command message - set event type to command and split data into command and args
+            event_type = EVENT_TELEGRAM_COMMAND
+            event_data.update(self._get_command_event_data(message.text))
+        else:
+            event_type = EVENT_TELEGRAM_TEXT
+            event_data[ATTR_TEXT] = message.text
+
+        if message.from_user:
+            event_data.update(
+                {
+                    ATTR_USER_ID: message.from_user.id,
+                    ATTR_FROM_FIRST: message.from_user.first_name,
+                    ATTR_FROM_LAST: message.from_user.last_name,
+                }
+            )
+
+        return event_type, event_data
+
+    def _get_callback_query_event_data(self, callback_query: CallbackQuery):
+        event_type = EVENT_TELEGRAM_CALLBACK
+        event_data: dict[str, Any] = {
+            ATTR_MSGID: callback_query.id,
+            ATTR_CHAT_INSTANCE: callback_query.chat_instance,
+            ATTR_DATA: callback_query.data,
+            ATTR_MSG: callback_query.message.to_dict()
+            if callback_query.message
+            else None,
```

Comment:
We prefer not using ternary operator expression if it spans more than one line. Please use a regular if/else block.